### DISCUSSION TITLE
Refer to other option during validation

### DISCUSSION THREAD

**Discussion by lppedd** [ajalt/clikt#467] [Q&A] [answered]
I've got a couple of tied options in my command:

```kotlin
private val basePath by option()
  .convert { it.toPath() }
  .defaultLazy { ".".toPath() }
  .validate {
    if (!it.isDirectory()) {
      fail("The base path '$it' does not exist or is not a valid directory")
    }
  }

private val configFile by option()
  .convert { it.toPath() }
  .defaultLazy { "./config.properties".toPath() }
  .validate {
    val tmpPath = basePath.resolve(it)

    if (!tmpPath.isRegularFile()) {
      fail("The configuration file '$tmpPath' does not exist or is not a valid file")
    }
  }
```

I need to validate `configFile` by resolving its name against the provided `basePath`.
Is the way I've done it now the correct way to do it? Or, is there an established way to accomplish the same?

Note the `.toPath()` comes from Okio multiplatform.

**Comment by ajalt** [comment, marked as answer]
`validate` is explicitly run [after everything has been parsed][1], so your approach is fine. The only thing I would change is to use the [built-in `path` conversion][2], which includes some validation already.

```kotlin
private val basePath by option()
    .path(mustExist = true, canBeFile = false)
    .default(Path("."))

private val configFile by option()
    .path()
    .default(Path("config.properties"))
    .validate {
        val tmpPath = basePath.resolve(it)
        if (!tmpPath.isRegularFile()) {
            fail("The configuration file '$tmpPath' does not exist or is not a valid file")
        }
    }
```

[1]: https://ajalt.github.io/clikt/parameters/#validate
[2]: https://ajalt.github.io/clikt/api/clikt/com.github.ajalt.clikt.parameters.types/path.html

**Reply by lppedd** [reply]
Thank you! Unfortunately I'm targeting native so the path options are not applicable. But thanks for the suggestion.
